Applications/WebPortal/WebApp/library/source/components/checkmark.tsx

```diff
@@ -0,0 +1,118 @@
+import { css, StyleSheet } from 'aphrodite';
+import * as React from 'react';
+import { DisplaySize } from '../display_size';
+
+interface Properties {
+
+  /** The size to display the component at. */
+  displaySize?: DisplaySize;
+
+  /** Determines if the box is checked. */
+  isChecked: boolean;
+
+  /** Determines if the component can be edited. */
+  readonly: boolean;
+
+  /** Called when the check mark is clicked on.
+   * @param newValue - The new value.
+   */
+  onClick?: (newValue?: boolean) => void;
+}
+
+/** A check mark component. */
+export class CheckMark extends React.Component<Properties> {
+  public static readonly defaultProps = {
+    readonly: false,
```

Comment:
What's the reasoning behind that? To make it more obvious which mode a component is in? 

I only ask because in HTML conventions readonly and disabled are usually optional.

Reply:
Boolean props are always optional in React by default, so you never have to explicitly mark it as such.

Reply:
Compiler seems to disagree with you. It seems like they need to be declared as optional but they don't seem to require default props.

Reply:
Do you have an example?

Reply:
If I change `readonly?: boolean;` to `readonly: boolean;` in _currency_selection_box.tsx_ the _RiskPageTester_ does not compile because `readonly` isn't a prop that's given to CurrencySelectionBox. CompliancePageTester is fine because the readonly property is explicitly passed as a prop for the compliance page stuff.

It is totally unnecessary to have `readonly` be in `defaultProps` though.

Reply:
Okay, then make ```boolean``` props optional for every component. We'll make it a coding standard that ```boolean``` props are always optional.